runtime/service_protocol.cc

```diff
@@ -53,21 +53,21 @@ ServiceProtocol::~ServiceProtocol() {
 
 void ServiceProtocol::AddHandler(Handler* handler,
                                  Handler::Description description) {
-  std::lock_guard<std::mutex> lock(handlers_mutex_);
+  std::unique_lock<std::shared_timed_mutex> lock(handlers_mutex_);
```

Comment:
I think `shared_timed_mutex` is C++14 only, can we use that here?

Reply:
this patch builds fine on windows too.